### PR TITLE
Add Robolectric to the project and port all tests

### PR DESCRIPTION
**Pull request by GabrielFleischer**

### CODE REVIEW

File: fe2-android/app/src/test/java/com/github/dedis/popstellar/testutils/Base64DataUtils.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+  /** @return a pseudo randomly generated signature */
+  public static Signature generateSignature() {
+    byte[] signature = new byte[SIGNATURE_LENGTH];
+    RANDOM.nextBytes(signature);
+    return new Signature(signature);
+  }
+
+  /** @return a pseudo randomly generated valid Ed25519 keypair */
+  public static KeyPair generateKeyPair() {
```
Comment (pierluca) [inline fe2-android/app/src/test/java/com/github/dedis/popstellar/testutils/Base64DataUtils.java:54]: I would highlight that this is highly insecure and only acceptable for tests, e.g. `generateInsecureKeyPair` , as we're not using a cryptographic random number generator.

File: fe2-android/app/src/test/java/com/github/dedis/popstellar/testutils/Base64DataUtils.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  /** @return a pseudo randomly generated valid Ed25519 keypair */
+  public static PoPToken generatePoPToken() {
+    byte[] privateKey = new byte[KEY_LENGTH];
+    RANDOM.nextBytes(privateKey);
+
+    Ed25519PrivateKeyParameters privateKeyParameters =
+        new Ed25519PrivateKeyParameters(privateKey, 0);
+    Ed25519PublicKeyParameters publicKeyParameters = privateKeyParameters.generatePublicKey();
+
+    return new PoPToken(privateKey, publicKeyParameters.getEncoded());
```
Comment (pierluca) [inline fe2-android/app/src/test/java/com/github/dedis/popstellar/testutils/Base64DataUtils.java:75]: why not just call `generateKeyPair` above? It should be nearly the same implementation, except for the return type.
Or vice versa, let `generateKeyPair` call this.

**Review by pierluca — COMMENTED** [review]
I went through everything, more or less in-depth.
It looks like some pretty solid work, though some code looks eerily familiar (e.g. the QR bitmap reader).
Isn't it massively duplicated from existing code? Is there really no way to factor out the shared parts ?

**Comment by GabrielFleischer** [comment]
I tried to create a shared source set, but I was not able to provide it the test dependencies.
I did not spend a lot of time, but as the goal is to completely get rid of the android tests, I did not want to spend multiple hours on something that will be deleted in a week.

File: fe2-android/app/src/test/java/com/github/dedis/popstellar/testutils/Base64DataUtils.java (hunk 1 above)
Comment (GabrielFleischer) [reply fe2-android/app/src/test/java/com/github/dedis/popstellar/testutils/Base64DataUtils.java]: Yes, this is a test-only utility class.

File: fe2-android/app/src/test/java/com/github/dedis/popstellar/testutils/Base64DataUtils.java (hunk 2 above)
Comment (GabrielFleischer) [reply fe2-android/app/src/test/java/com/github/dedis/popstellar/testutils/Base64DataUtils.java]: Yes, this has been addressed

File: fe2-android/app/src/test/java/com/github/dedis/popstellar/testutils/UITestUtils.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+public class UITestUtils {
+
+  /**
+   * Assert that the latest toast showed the provided text
+   *
+   * @param resId resource of the text
+   * @param args arguments to the resource
+   */
+  public static void assertToastIsShown(@StringRes int resId, Object... args) {
+    assertThat("No toast have been displayed", ShadowToast.getLatestToast(), notNullValue());
```
Comment (pierluca) [inline fe2-android/app/src/test/java/com/github/dedis/popstellar/testutils/UITestUtils.java:30]: The message doesn't seem to align with the function title 🤔 

Also, English fix, if this is actually what you want to say :)
```suggestion
    assertThat("No toast has been displayed", ShadowToast.getLatestToast(), notNullValue());
```
or
```suggestion
    assertThat("No toast was displayed", ShadowToast.getLatestToast(), notNullValue());
```

File: fe2-android/app/src/test/java/com/github/dedis/popstellar/testutils/UITestUtils.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  /**
+   * Retrieve the last opened dialog expecting it to be of the provided type
+   *
+   * @param type of the dialog
+   * @param <D> of the dialog
+   * @return the dialog
+   */
+  public static <D extends Dialog> D getLastDialog(Class<D> type) {
+    Dialog dialog = ShadowDialog.getLatestDialog();
+    assertThat("No dialog have been displayed", dialog, notNullValue());
```
Comment (pierluca) [inline fe2-android/app/src/test/java/com/github/dedis/popstellar/testutils/UITestUtils.java:45]:
```suggestion
    assertThat("No dialog has been displayed", dialog, notNullValue());
```

**Review by pierluca — APPROVED** [review]

File: fe2-android/app/src/test/java/com/github/dedis/popstellar/testutils/UITestUtils.java (hunk 3 above)
Comment (GabrielFleischer) [reply fe2-android/app/src/test/java/com/github/dedis/popstellar/testutils/UITestUtils.java:30]: Would `assertToastIsDisplayedWithText()` be better ?

**Review by pierluca — APPROVED** [review]

**Pull request merged**